.github/workflows/checks-and-tests.yml

```diff
@@ -0,0 +1,62 @@
+name: Checks and Tests
+on:
+  pull_request:
+    types: [opened, reopened, synchronize]
+  workflow_dispatch:
+jobs:
+  checks-and-tests:
```

Comment:
Sorry but I got more questions :see_no_evil: 

I see that you've bundled together 3 actions into one now.
Won't this make the overall process take longer?
We could run UTs and build in parallel, right?

It only takes a few minutes anyhow.

Reply:
It seems parallel is faster
In pararell, `6m 59s` for test and `4m 8s` for checks. -> `6m 59s`
In sequential, `5m 13s` for test and `3m 47s` for checks. -> `9m`